JitsiConferenceEventManager.js

```diff
@@ -0,0 +1,495 @@
+/* global Strophe */
+var logger = require("jitsi-meet-logger").getLogger(__filename);
+var EventEmitterForwarder = require("./modules/util/EventEmitterForwarder");
+var XMPPEvents = require("./service/xmpp/XMPPEvents");
+var RTCEvents = require("./service/RTC/RTCEvents");
+var JitsiConferenceEvents = require("./JitsiConferenceEvents");
+var JitsiConferenceErrors = require("./JitsiConferenceErrors");
+var AuthenticationEvents =
+    require("./service/authentication/AuthenticationEvents");
+var Statistics = require("./modules/statistics/statistics");
+var MediaType = require("./service/RTC/MediaType");
+
+/**
+ * Setups all event listeners related to conference
+ * @param conference {JitsiConference} the conference
+ */
+function JitsiConferenceEventManager(conference) {
+    this.conference = conference;
+    this.chatRoomForwarder = new EventEmitterForwarder(this.conference.room,
+        this.conference.eventEmitter);
+}
+
+/**
+ * Setups event listeners related to conference.chatRoom
+ */
+JitsiConferenceEventManager.prototype.setupChatRoomListeners = function () {
+    var conference = this.conference;
+
+    conference.room.addListener(XMPPEvents.ICE_RESTARTING, function () {
```

Comment:
conference.room is going to appear in the following source code a number of times. I think it makes sense to reduce the complexity of the source code by defining a, for example, var chatRoom = conference.room; just like var conference = this.conference;